types/index.d.ts

```diff
@@ -253,31 +247,34 @@ declare namespace unified {
    * A pairing of a plugin with its settings
    *
    * @typeParam S Plugin settings
-   * @typeParam S2 Extra plugin settings
    * @typeParam P Processor settings
    */
-  type PluginTuple<S = Settings, S2 = undefined, P = Settings> =
-    | [Plugin<S, undefined, P>, S]
-    | [Plugin<S, S2, P>, S, S2]
+  type PluginTuple<S extends any[] = [Settings?], P = Settings> = [
+    Plugin<S, P>,
+    /**
+     * NOTE: ideally this would be S instead of any[]
+     * As of TypeScript 3.5.2 generic tuples cannot be spread
+     */
+    ...any[]
```

Comment:
Trying to use `S` here currently results in
```
A rest element type must be an array type
```

This may be a bug, it may be a feature that is not implemented.

---

The trade off here, is we lose variadic tuple type checking of settings, when using tuple syntax (mildly ironic :stuck_out_tongue_winking_eye: ).
This is reflected in the tests where
```ts
processor.use([typedPlugin, invalidSettings])
```

is not caught by TypeScript.

Reply:
> ```
> A rest element type must be an array type
> ```

There is an issue upstream tracking this https://github.com/microsoft/TypeScript/issues/26113